src/Nethermind/Nethermind.Trie/VerkleTree.cs

```diff
@@ -265,12 +265,22 @@ public void SetCode(Address address, byte[] code)
         //     byte[] chunkKey = GetTreeKeyForCodeChunk(address, (UInt256)i);
         //     SetValue(chunkKey, processedCode.Slice(i*32, 32));
         // }
-
-        int i = 0;
+        
+        UInt256 i = 0;
+        Span<byte> subIndexBytes = stackalloc byte[32];
         CodeChunkEnumerator chunkEnumerator = new(code);
         while (chunkEnumerator.TryGetNextChunk(out byte[] chunk))
         {
-            byte[] chunkKey = GetTreeKeyForCodeChunk(address, (UInt256)i);
+            // byte[] chunkKey = GetTreeKeyForCodeChunk(address, (UInt256)i);
+            
+            // find tree index and the sub index in verkle tree for the code chunk
+            UInt256 chunkOffset = CodeOffset + i;
+            UInt256 treeIndex = chunkOffset / VerkleNodeWidth;
+            UInt256.Mod(chunkOffset, VerkleNodeWidth, out UInt256 subIndex);
+            subIndex.ToBigEndian(subIndexBytes);
```

Comment:
This probably could/should be in separate method, like FillTreeKeyForCodeChunk taking Span